proxy/src/main/java/com/velocitypowered/proxy/connection/client/ClientPlaySessionHandler.java

```diff
@@ -229,9 +230,8 @@ public boolean handle(PluginMessage packet) {
         player.getKnownChannels().removeAll(PluginMessageUtil.getChannels(packet));
         backendConn.write(packet.retain());
       } else if (PluginMessageUtil.isMcBrand(packet)) {
-        player.setClientBrand(PluginMessageUtil.readBrandMessage(packet.content()));
-        backendConn.write(PluginMessageUtil
-            .rewriteMinecraftBrand(packet, server.getVersion(), player.getProtocolVersion()));
+        String brand = PluginMessageUtil.readBrandMessage(packet.content());
+        server.getEventManager().fire(new PlayerClientBrandEvent(player, brand));
```

Comment:
After the event is fired, player.setClientBrand and backendConn.write should still be called.

Reply:
whoops those accidentally disappeared in me making it non-cancelable!